internal/kafka/internal/services/quota/quota_management_list_service.go

```diff
@@ -95,28 +123,75 @@ func (q QuotaManagementListService) ReserveQuota(kafka *dbapi.KafkaRequest, inst
 		totalInstanceCount += kafkaInstanceSize.CapacityConsumed
 	}
 
-	if quotaManagementListItem != nil && instanceType == types.STANDARD {
+	if quotaManagementListItem != nil && kafka.InstanceType != types.DEVELOPER.String() {
 		kafkaInstanceSize, e := q.kafkaConfig.GetKafkaInstanceSize(kafka.InstanceType, kafka.SizeId)
 		if e != nil {
 			return "", errors.NewWithCause(errors.ErrorGeneral, e, "Error reserving quota")
 		}
-		if quotaManagementListItem.IsInstanceCountWithinLimit(totalInstanceCount + kafkaInstanceSize.CapacityConsumed) {
+		if quotaManagementListItem.IsInstanceCountWithinLimit(kafka.InstanceType, billingModelID, totalInstanceCount+kafkaInstanceSize.CapacityConsumed) {
+			kafka.ActualKafkaBillingModel = kafka.DesiredKafkaBillingModel
 			return "", nil
 		} else {
 			return "", errors.MaximumAllowedInstanceReached(message)
 		}
 	}
 
-	if instanceType == types.DEVELOPER && quotaManagementListItem == nil {
+	if kafka.InstanceType == types.DEVELOPER.String() && quotaManagementListItem == nil {
 		if totalInstanceCount >= quota_management.GetDefaultMaxAllowedInstances() {
 			return "", errors.MaximumAllowedInstanceReached(message)
 		}
+		kafka.ActualKafkaBillingModel = kafka.DesiredKafkaBillingModel
 		return "", nil
 	}
 
 	return "", errors.InsufficientQuotaError("Insufficient Quota")
 }
 
+func (q QuotaManagementListService) detectBillingModel(kafka *dbapi.KafkaRequest) (string, *errors.ServiceError) {
+	if kafka.DesiredKafkaBillingModel != "" {
+		return kafka.DesiredKafkaBillingModel, nil
+	}
+	if kafka.InstanceType == types.DEVELOPER.String() {
+		return billingModelStandard, nil
```

Comment:
Side comment: This assumes that developer is defined as an instance type in the instances type configuration, and also assumes than standard exists for developer types